server/src/main/java/org/apache/druid/client/DirectDruidClient.java

```diff
@@ -494,6 +504,12 @@ public void onFailure(Throwable t)
           Execs.directExecutor()
       );
     }
+    catch (RuntimeException e) {
+      if (openConnections.get() > 0) {
+        openConnections.getAndDecrement();
+      }
+      throw e;
+    }
```

Comment:
This could be moved to after `httpClient.go` call. Also, why not decrement the counter for any exception and why is there a check on `get() > 0`

Reply:
added just to be on safe side, But it will never happen. Will remove the condition.